pkg/resource/metadata.go

```diff
@@ -172,19 +175,29 @@ func (md Metadata) Equal(other Metadata) bool {
 		return true
 	}
 
-	fins := append(Finalizers(nil), md.fins...)
-	otherFins := append(Finalizers(nil), other.fins...)
-
-	sort.Strings(fins)
-	sort.Strings(otherFins)
-
-	for i := range fins {
-		if fins[i] != otherFins[i] {
-			return false
+	switch len(md.fins) {
+	case 0:
+		return true
+	case 1:
+		return md.fins[0] == other.fins[0]
+	case 2:
+		return md.fins[0] == other.fins[0] && md.fins[1] == other.fins[1] ||
+			md.fins[0] == other.fins[1] && md.fins[1] == other.fins[0]
+	default:
+		fins := slices.Clone(md.fins)
+		otherFins := slices.Clone(other.fins)
+
+		sort.Strings(fins)
+		sort.Strings(otherFins)
```

Comment:
```suggestion
		slices.Sort(fins)
		slices.Sort(otherFins)
```

Generic version of sort should be faster.

Reply:
updated